Tests fail when built with go 1.26.2

I'm getting the following failures when using go 1.26.2. I can reproduce with git c78a972ea2721e693d910f0cf9a10d2ba411f69d but also 0.197.0 release. The tests pass after downgrading to go 1.25.9.

```console
$ go version
go version go1.26.2-X:nodwarf5 linux/amd64
$ make build
go build -mod=vendor -o stripe-mock
$ make test
# -count=1 disables the cache
go test ./... -count=1
ok  	github.com/stripe/stripe-mock	0.015s
?   	github.com/stripe/stripe-mock/embedded	[no test files]
ok  	github.com/stripe/stripe-mock/generator/datareplacer	0.012s
ok  	github.com/stripe/stripe-mock/param	0.011s
ok  	github.com/stripe/stripe-mock/param/coercer	0.014s
ok  	github.com/stripe/stripe-mock/param/form	0.005s [no tests to run]
ok  	github.com/stripe/stripe-mock/param/nestedtypeassembler	0.019s
ok  	github.com/stripe/stripe-mock/param/parser	0.018s
--- FAIL: TestDoubleSlashFixHandler (0.00s)
        Error Trace:    server_test.go:339
        Error:		Not equal: 301 (expected)
        	        != 307 (actual)
        
Routing to 7 path(s) and 8 endpoint(s) with 7 validator(s)
Request: POST /v1/charges
Request data = map[amount:123]
Response: elapsed=49.053µs status=200
Routing to 7 path(s) and 8 endpoint(s) with 7 validator(s)
Request: POST /v1/charges
Request data = map[]
Request validation error: validator 0x1b453dd639e0 failed: object property 'amount' is required
Response: elapsed=71.966µs status=400
Routing to 7 path(s) and 8 endpoint(s) with 7 validator(s)
Request: POST /v1/charges
Request data = map[amount:123 doesntexist:foo]
Request validation error: validator 0x1b453e62ed80 failed: additional properties are not allowed
Response: elapsed=26.571µs status=400
Routing to 7 path(s) and 8 endpoint(s) with 7 validator(s)
Request: GET /v1/charges
Request data = map[limit:10]
Response: elapsed=27.352µs status=200
Routing to 7 path(s) and 8 endpoint(s) with 7 validator(s)
Request: GET /v1/charges
Request data = map[doesntexist:foo limit:10]
Request validation error: validator 0x1b453e1d90b0 failed: additional properties are not allowed
Response: elapsed=25.659µs status=400
Routing to 7 path(s) and 8 endpoint(s) with 7 validator(s)
Request: GET /a
Response: elapsed=6.673µs status=401
Routing to 7 path(s) and 8 endpoint(s) with 7 validator(s)
Request: GET /v1/charges
Response: elapsed=8.997µs status=400
Routing to 7 path(s) and 8 endpoint(s) with 7 validator(s)
Request: GET /v1/charges
Request data = map[]
Response: elapsed=18.024µs status=200
Routing to 7 path(s) and 8 endpoint(s) with 7 validator(s)
Request: POST /v1/charges
Request data = map[amount:123]
Response: elapsed=70.334µs status=200
Routing to 7 path(s) and 8 endpoint(s) with 7 validator(s)
Request: POST /
Response: elapsed=10.079µs status=401
Routing to 7 path(s) and 8 endpoint(s) with 7 validator(s)
Request: POST /
Response: elapsed=7.364µs status=404
Routing to 7 path(s) and 8 endpoint(s) with 7 validator(s)
Request: POST /v1/charges
Request data = map[]
Request validation error: validator 0x1b453eeb9c50 failed: object property 'amount' is required
Response: elapsed=20.168µs status=400
Routing to 7 path(s) and 8 endpoint(s) with 7 validator(s)
Request: POST /v1/charges
Request data = map[amount:123]
Response: elapsed=28.594µs status=200
Routing to 7 path(s) and 8 endpoint(s) with 7 validator(s)
Request: POST /v1/charges
Request's `Content-Type` header was empty. Expected: `application/x-www-form-urlencoded`.
Response: elapsed=12.314µs status=400
Routing to 7 path(s) and 8 endpoint(s) with 7 validator(s)
Request: DELETE /v1/customers/cus_123
Request data = map[]
Response: elapsed=16.371µs status=200
Routing to 7 path(s) and 8 endpoint(s) with 7 validator(s)
Request: POST /v1/charges
Request's `Content-Type` didn't match the path's expected media type. Expected: `application/x-www-form-urlencoded`. Was: `application/json`.
Response: elapsed=12.053µs status=400
body = {"error":{"message":"Request's `Content-Type` didn't match the path's expected media type. Expected: `application/x-www-form-urlencoded`. Was: `application/json`.","type":"invalid_request_error"}}
Routing to 7 path(s) and 8 endpoint(s) with 7 validator(s)
Request: POST /v1/charges
Request data = map[amount:123]
Response: elapsed=24.887µs status=200
Routing to 7 path(s) and 8 endpoint(s) with 7 validator(s)
Routing to 7 path(s) and 8 endpoint(s) with 7 validator(s)
Request: GET /v1/charges
Request data = map[]
Response: elapsed=20.299µs status=200
Routing to 7 path(s) and 8 endpoint(s) with 7 validator(s)
Request: GET /v1/quotes/qt_123/pdf
Request data = map[]
Response: elapsed=10.42µs status=200
FAIL
FAIL	github.com/stripe/stripe-mock/server	34.471s
ok  	github.com/stripe/stripe-mock/spec	0.006s
FAIL
make: *** [Makefile:18: test] Error 1

```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -6,12 +6,16 @@ permissions: {}
 
 jobs:
   test:
-    name: Test
+    name: Test (Go ${{ matrix.go-version }})
 
     runs-on: ubuntu-latest
 
     permissions:
       contents: read
+
+    strategy:
+      matrix:
+        go-version: ["1.25", "1.26"]
 
     steps:
       - uses: actions/checkout@v2
@@ -21,7 +25,7 @@ jobs:
       - name: Set up Go
         uses: actions/setup-go@v2
         with:
-          go-version: 1.22.1
+          go-version: ${{ matrix.go-version }}
       - name: test
         run: |
           go install golang.org/x/lint/golint@latest &&

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -19,7 +19,7 @@ jobs:
           fetch-depth: 0
           submodules: recursive
       - name: Set up Go
-        uses: actions/setup-go@v2
+        uses: actions/setup-go@v6
         with:
           go-version: stable
       - name: test

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -6,16 +6,12 @@ permissions: {}
 
 jobs:
   test:
-    name: Test (Go ${{ matrix.go-version }})
+    name: Test
 
     runs-on: ubuntu-latest
 
     permissions:
       contents: read
-
-    strategy:
-      matrix:
-        go-version: ["1.25", "1.26"]
 
     steps:
       - uses: actions/checkout@v2
@@ -25,7 +21,7 @@ jobs:
       - name: Set up Go
         uses: actions/setup-go@v2
         with:
-          go-version: ${{ matrix.go-version }}
+          go-version: stable
       - name: test
         run: |
           go install golang.org/x/lint/golint@latest &&

--- a/server/server_test.go
+++ b/server/server_test.go
@@ -5,7 +5,6 @@ import (
 	"encoding/base64"
 	"encoding/json"
 	"fmt"
-	"go/version"
 	"io/ioutil"
 	"net/http"
 	"net/http/httptest"
@@ -323,30 +322,6 @@ func TestDoubleSlashFixHandler(t *testing.T) {
 
 		assert.Equal(t, http.StatusOK, w.Code)
 		assert.Equal(t, "/v1/charges", lastPath)
-	}
-
-	// Demonstrates the (undesirable) standard Go behavior without the handler
-	{
-		lastPath = ""
-
-		req := httptest.NewRequest(
-			http.MethodGet, "http://example.com//v1/charges", nil)
-		w := httptest.NewRecorder()
-
-		// Note that we skip the double slash fix handler and have the mux
-		// serve directly
-		httpMux.ServeHTTP(w, req)
-
-		// Go 1.26 changed this redirect from 301 to 307 to preserve the HTTP
-		// method. Both indicate the mux redirected rather than serving the request.
-		// see: https://go.dev/doc/go1.26#nethttppkgnethttp
-		if version.Compare(runtime.Version(), "go1.26") >= 0 {
-			assert.Equal(t, http.StatusTemporaryRedirect, w.Code)
-		} else {
-			assert.Equal(t, http.StatusMovedPermanently, w.Code)
-		}
-
-		assert.Equal(t, "", lastPath)
 	}
 }
 

--- a/server/server_test.go
+++ b/server/server_test.go
@@ -5,7 +5,7 @@ import (
 	"encoding/base64"
 	"encoding/json"
 	"fmt"
-	"github.com/stripe/stripe-mock/embedded"
+	"go/version"
 	"io/ioutil"
 	"net/http"
 	"net/http/httptest"
@@ -13,6 +13,8 @@ import (
 	"path"
 	"runtime"
 	"testing"
+
+	"github.com/stripe/stripe-mock/embedded"
 
 	assert "github.com/stretchr/testify/require"
 	"github.com/stripe/stripe-mock/spec"
@@ -335,8 +337,15 @@ func TestDoubleSlashFixHandler(t *testing.T) {
 		// serve directly
 		httpMux.ServeHTTP(w, req)
 
-		// This is the default Go behavior (301)
-		assert.Equal(t, http.StatusMovedPermanently, w.Code)
+		// Go 1.26 changed this redirect from 301 to 307 to preserve the HTTP
+		// method. Both indicate the mux redirected rather than serving the request.
+		// see: https://go.dev/doc/go1.26#nethttppkgnethttp
+		if version.Compare(runtime.Version(), "go1.26") >= 0 {
+			assert.Equal(t, http.StatusTemporaryRedirect, w.Code)
+		} else {
+			assert.Equal(t, http.StatusMovedPermanently, w.Code)
+		}
+
 		assert.Equal(t, "", lastPath)
 	}
 }

